src/Database/PG/Query/PTI.hs

```diff
@@ -1,161 +1,296 @@
-{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE PatternSynonyms #-}
 
-module Database.PG.Query.PTI where
+{-
+Description: Oid constants.
 
--------------------------------------------------------------------------------
+All these are taken from this declaration file in the postgres repo:
+https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_type.dat
+-}
+module Database.PG.Query.PTI where
 
-import Data.Word (Word32)
 import Database.PostgreSQL.LibPQ qualified as PQ
-import Prelude
-
--------------------------------------------------------------------------------
-
-mkOid :: Word32 -> PQ.Oid
-mkOid = PQ.Oid . fromIntegral
 
 -- * Constants
+--
+-- We use pattern synonyms for the Oids so that we can pattern-match on them.
 
--------------------------
-auto = mkOid 0
+pattern Auto :: PQ.Oid
+pattern Auto = PQ.Oid 0
 
-abstime = mkOid 702
+pattern Abstime :: PQ.Oid
+pattern Abstime = PQ.Oid 702
 
-aclitem = mkOid 1033
+pattern Aclitem :: PQ.Oid
+pattern Aclitem = PQ.Oid 1033
 
-bit = mkOid 1560
+pattern Bit :: PQ.Oid
+pattern Bit = PQ.Oid 1560
 
-bool = mkOid 16
+pattern Bool :: PQ.Oid
+pattern Bool = PQ.Oid 16
 
-box = mkOid 603
+pattern Box :: PQ.Oid
+pattern Box = PQ.Oid 603
 
-bpchar = mkOid 1042
+pattern Bpchar :: PQ.Oid
+pattern Bpchar = PQ.Oid 1042
 
-bytea = mkOid 17
+pattern Bytea :: PQ.Oid
+pattern Bytea = PQ.Oid 17
 
-char = mkOid 18
+pattern Char :: PQ.Oid
+pattern Char = PQ.Oid 18
 
-cid = mkOid 29
+pattern Cid :: PQ.Oid
+pattern Cid = PQ.Oid 29
 
-cidr = mkOid 650
+pattern Cidr :: PQ.Oid
+pattern Cidr = PQ.Oid 650
 
-circle = mkOid 718
+pattern Circle :: PQ.Oid
+pattern Circle = PQ.Oid 718
 
-cstring = mkOid 2275
+pattern Cstring :: PQ.Oid
+pattern Cstring = PQ.Oid 2275
 
-date = mkOid 1082
+pattern Date :: PQ.Oid
+pattern Date = PQ.Oid 1082
 
-daterange = mkOid 3912
+pattern Daterange :: PQ.Oid
+pattern Daterange = PQ.Oid 3912
 
-float4 = mkOid 700
+pattern Float4 :: PQ.Oid
+pattern Float4 = PQ.Oid 700
 
-float8 = mkOid 701
+pattern Float8 :: PQ.Oid
+pattern Float8 = PQ.Oid 701
 
-gtsvector = mkOid 3642
+pattern Gtsvector :: PQ.Oid
+pattern Gtsvector = PQ.Oid 3642
 
-inet = mkOid 869
+pattern Inet :: PQ.Oid
+pattern Inet = PQ.Oid 869
 
-int2 = mkOid 21
+pattern Int2 :: PQ.Oid
+pattern Int2 = PQ.Oid 21
 
-int2vector = mkOid 22
+pattern Int2vector :: PQ.Oid
+pattern Int2vector = PQ.Oid 22
 
-int4 = mkOid 23
+pattern Int4 :: PQ.Oid
+pattern Int4 = PQ.Oid 23
 
-int4range = mkOid 3904
+pattern Int4range :: PQ.Oid
+pattern Int4range = PQ.Oid 3904
 
-int8 = mkOid 20
+pattern Int8 :: PQ.Oid
+pattern Int8 = PQ.Oid 20
 
-int8range = mkOid 3926
+pattern Int8range :: PQ.Oid
+pattern Int8range = PQ.Oid 3926
 
-interval = mkOid 1186
+pattern Interval :: PQ.Oid
+pattern Interval = PQ.Oid 1186
 
-json = mkOid 114
+pattern Json :: PQ.Oid
+pattern Json = PQ.Oid 114
 
-jsonb = mkOid 3802
+pattern Jsonb :: PQ.Oid
+pattern Jsonb = PQ.Oid 3802
```

Comment:
Very small nit: I think I'd prefer those (and others) with proper capitalization? `JSON`, `JSONB`, `MacAddr`...